src/segment.rs

```diff
@@ -643,6 +643,22 @@ impl Segment {
             }
         }
     }
+
+    pub fn save_to_path<P>(&self, path: P) -> Result<()>
```

Comment:
To guarantee safety, we might want to reject if a file already exists on that path.

So I recommend to check this and return with an error if it is already there.

The `as_mut_slice()` function is unsafe because it cannot guarantee this itself.

Reply:
Added file existing check. Also added such checks on higher level (`lib.rs`)